sentry-core/src/main/java/io/sentry/protocol/Device.java

```diff
@@ -1,5 +1,239 @@
 package io.sentry.protocol;
 
+import java.util.Date;
+import java.util.TimeZone;
+
 public class Device {
-  static String TYPE = "device";
+  static final String TYPE = "device";
+
+  private String name;
+  private String manufacturer;
+  private String brand;
+  private String family;
+  private String model;
+  private String modelId;
+  private String architecture;
+  private Short batteryLevel;
+  private Boolean isCharging;
```

Comment:
is there a reason why always using Objects instead of primitive types?

Reply:
A primitive type can't be null so is charging by default would be `false` and if that was explicitly set or not would be unknown.
The protocol is composed of _all optional_ fields so it's important that by default everything is null. Serializing an instance of `SentryEvent` where no setter was called should result in a JSON with `event_id` and `timestamp` only even though the server would accept it even if it was simply `{}`.

Reply:
got it, makes sense, thanks